### PR TITLE
docs: Add modal_show/remove examples

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes https://github.com/posit-dev/py-shiny-site/issues/201
Related: #1318

Adds an example using `ui.modal_show()` and `ui.modal_remove()` that is re-used in the documentation for both functions.

### CODE REVIEW

File: shiny/api-examples/modal_remove/app-express.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    )
+
+
+@reactive.effect
+@reactive.event(input.run)
+def do_run_model():
+    # Show the modal, blocking interaction with the UI
+    ui.modal_show(the_modal())
+
+    result = run_model(delay=2)
```
Comment (gadenbuie) [inline shiny/api-examples/modal_remove/app-express.py:35]:
```suggestion
    result = run_model(delay=4)
```

File: shiny/api-examples/modal_remove/app-core.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def do_run_model():
+        # Show the modal, blocking interaction with the UI
+        ui.modal_show(the_modal())
+
+        result = run_model(delay=4)
+
+        # Now that we have model results, remove the modal
+        # and update the model result reactive value
+        ui.modal_remove()
+        model_result.set(result)
```
[the remark this answers is not in the document]
Comment (gadenbuie) [reply shiny/api-examples/modal_remove/app-core.py:42]: I don't think we should get rid of it; it's a core part of the pattern. On the other hand I don't think it's worth showing the value; that ends up just being something to delete if you use this example as a template.

File: shiny/ui/_modal.py [hunk 3]
```diff
@@ -158,7 +158,7 @@ def modal(
     )
 
 
-@add_example(ex_dir="../api-examples/modal")
+@add_example(ex_dir="../api-examples/modal_remove")
 def modal_show(modal: Tag, session: Optional[Session] = None) -> None:
```
Comment (cpsievert) [inline shiny/ui/_modal.py:162]: I'd prefer keeping the simpler `../api-examples/modal` example for `modal_show()`. In many cases, folks just need to show and not remove programmatically.

File: shiny/ui/_modal.py [hunk 4]
```diff
@@ -158,7 +158,7 @@ def modal(
     )
 
 
-@add_example(ex_dir="../api-examples/modal")
+@add_example(ex_dir="../api-examples/modal_remove")
```
Comment (gadenbuie) [inline shiny/ui/_modal.py:161]:
```suggestion
@add_example(ex_dir="../api-examples/modal")
```

File: shiny/ui/_modal.py (hunk 3 above)
Comment (gadenbuie) [reply shiny/ui/_modal.py:162]: sure that makes sense!

File: shiny/api-examples/modal_remove/app-core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpsievert) [reply shiny/api-examples/modal_remove/app-core.py:42]: > it's a core part of the pattern

I'd say it's core part of _a_ pattern, but not everyone coming to `modal_remove()` wants that pattern? IMO introducing state that isn't actually used has potential to cause confusion.

**Review by cpsievert — APPROVED** [review]

File: shiny/api-examples/modal_remove/app-core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gadenbuie) [reply shiny/api-examples/modal_remove/app-core.py:42]: > It's a core part of a pattern, but not everyone coming to modal_remove() wants that pattern

lol very true. I meant "the pattern being demonstrated", not the _only_ pattern. 

> IMO introducing state that isn't actually used has potential to cause confusion. 

I probably wouldn't have included it either, but I've independently shown this pattern to 2-4 different people asking about how to remove the modal.

**Pull request merged**